internal/session/query.go

```diff
@@ -422,6 +422,69 @@ update session_credential
 		key_id = ?
 where session_id = ?
 	and credential_sha256 = ?;
+`
+	listSessionsTemplate = `
+with session_ids as (
+	select public_id
+	  from session
+	   -- search condition for applying permissions is constructed
+	 where %s
+  order by create_time desc, public_id asc
+	 limit %d
+)
+   select *
+     from session_list
+    where session_list.public_id in (select * from session_ids)
+ order by create_time desc, public_id asc;
```

Comment:
It took me a while to see it, but I like @tmessi's query formatting - in this CTE, we start by indenting the `order by` by 2 spaces, and then align all the other keywords (select, from, where, limit) with the end of that string, so you end up with a vertical space between key words and values. I think maybe yours was correct but with tabs instead of spaces, which I think we should avoid in string literals.
```suggestion
with session_ids as (
    select public_id
      from session
           -- search condition for applying permissions is constructed
     where %s
  order by create_time desc, public_id asc
     limit %d
)
   select *
     from session_list
    where session_list.public_id in (select * from session_ids)
 order by create_time desc, public_id asc;
```